### PR TITLE
Fax Machines

### PR DESCRIPTION
**Pull request by Cenrus**
## About The Pull Request

Adds Fax Machines from Oracle station, original PR by top secret shadow government dictator ike709. Lawyers and the captain will get a long ranged fax machine which can be used to message CentCom.

TGUI 3.0 Interface done by Mark Suckerberg (thank you)

## Why It's Good For The Game

Encourages usage of paperwork, Roleplay opportunities, requested by our downstream.

## Changelog
:cl: ike709, Mark Suckerberg for the UI
add: Fax machines from Oracle
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
epic

**Comment by ike709** [comment]
Show a pic of the UI.

File: code/__HELPERS/unique_ids.dm
Comment (ike709) [inline code/__HELPERS/unique_ids.dm]: Please don't use a colon and instead just check like the comment says.

**Comment by Cenrus** [comment]
image

**Comment by ghost** [comment]
I swear if this is para/oracle-fax code I'm gonna...

OF COURSE IT IS. UGGHHHHHH

**Review by ghost — CHANGES_REQUESTED** [review]
Para/oracle fax code is HOT GARBAGE. STOP FUCKING USING IT.

**Comment by Cenrus** [comment]
Good feature but not worth the effort for a full rewrite.

**Comment by zeskorion** [comment]
> Para/oracle fax code is HOT GARBAGE. STOP FUCKING USING IT.

says a hippie coder

File: code/modules/paperwork/paper.dm
Comment (zeskorion) [inline code/modules/paperwork/paper.dm:457]: is this just an admin only thing?

**Review by zeskorion — DISMISSED** [review]

**Comment by ghost** [comment]
> 
> 
> > Para/oracle fax code is HOT GARBAGE. STOP FUCKING USING IT.
> 
> says a hippie coder

a good hippie coder

File: code/modules/paperwork/paper.dm
Comment (Cenrus) [reply code/modules/paperwork/paper.dm:457]: Yep. If someone faxes Centcom WGW or something similarly stupid, admins can smite them with these.

**Comment by MCterra10** [comment]
"why are you booing me? I'm right!"

File: code/modules/paperwork/paper.dm
Comment (Useroth) [inline code/modules/paperwork/paper.dm]:
```suggestion
			. = ..()
```
I'm not sure about this codebase, but it runtimes on mine with the lone `..()` as opposed to `. = ..()`

Comment (Cenrus) [reply code/modules/paperwork/paper.dm]: Aye, I'm sure this is because of the examine code refactor we merged a while ago.

**Comment by Cenrus** [comment]
Build error is caused by an atmos runtime

**Comment by ike709** [comment]
Looks like meta and pubby are no longer TGM, going by the diff count.

**Comment by Cenrus** [comment]
Map hasn't changed, the diffs are just acting up. It's like BeeStation.dme all over again.

**Comment by Cenrus** [comment]
Fixed maps

**Comment by Cenrus** [comment]
ready

**Comment by Cenrus** [comment]
Status on the merge? NSV are doing an upstream pull soon.

**Review by PowerfulBacon — COMMENTED** [review]
Seems to generally work ok.
Lots of whitespace.
Another thing I noticed is that there is no cooldown on sending faxes, so you can spam the button to make the receiving fax machine play really annoying noises (although it doesn't spam the paper).
Also what is the 'Internal Affairs' department?

**Comment by Cenrus** [comment]
Internal Affairs is the lawyer's office.

**Comment by ike709** [comment]
IA should be renamed. It was called that on oracle because IA was an actual role. Just name it Law Office or something.

**Comment by Cenrus** [comment]
It's called law office now

**Comment by Cenrus** [comment]
Sound was removed, too

**Comment by eLeCtrOssSnake** [comment]
Cringe useless item, captain won't listen to you even if you lynch him... What fax are you even talking about.

**Comment by ike709** [comment]
The reason we've been avoiding this PR is because we've been waiting on the paper revamp from TG. That's now ported: https://github.com/BeeStation/BeeStation-Hornet/pull/2035

Please make faxes work with markdown paper and we'll give this a look/testmerge.

**Comment by Kmc2000** [comment]
Nice! I want this

**Comment by Cenrus** [comment]
going to look at this after markdown paper is merged

**Comment by ToasterBan** [comment]
lawyers will have actual power? fuck yeah cant wait till this gets merged

**Comment by TeomanTheGreat** [comment]
Wow noone is reviewing this

**Comment by imw-Oculus** [comment]
I'll never downvote a new feature. The idea is good.

**Pull request closed without merging**
